src/numeric_index.c

```diff
@@ -899,8 +874,8 @@ void NumericRangeIterator_OnReopen(void *privdata) {
   }
 
   if (it->type == READ_ITERATOR) {
-    IndexReader_Reopen(it->ctx, nu);
+    IndexReader_OnReopen(it->ctx);
   } else if (it->type == UNION_ITERATOR) {
-    UI_Foreach(it, IndexReader_Reopen, nu);
+    UI_Foreach(it, IndexReader_OnReopen);
```

Comment:
A union of unions is not possible? I recall that you mentioned something about this case being not taken care of

Reply:
On geo queries there is 0-9 options to check (related to the hashing method) and in the “worst” case we have a union of 9 numeric ranges, each one could be a union of index readers. But each numeric range is registered separately to “onReopen” so here there is no other possibility. We could add here a log assert if it’s neither of these cases

Reply:
then lets do it